### DISCUSSION TITLE
Displaying Loading Page While Fetching Data in Nuxt

### DISCUSSION THREAD

**Discussion by oxince** [nuxt/nuxt#22502] [Questions]
Today, I embarked on creating my inaugural Nuxt project and found myself pondering over the intricacies of when the loading page should make its appearance. To address this, I introduced a promise in my API, inducing a deliberate 4000ms delay. Concurrently, I employed the "useFetch" function on my index page.

Yet, to my chagrin, the anticipated loading screen failed to materialize during the 4000ms wait time imposed by the API. Instead, all I observed was the generic browser loading indication.

My objective is to seamlessly exhibit the Loading component while the data fetching process is underway. Regrettably, at present, the Loading component only surfaces after the aforementioned 4000ms interval has elapsed.

**Comment by warflash** [comment]
Set lazy to true in useFetch options https://nuxt.com/docs/api/composables/use-fetch#:~:text=%3F%3A%20boolean-,lazy%3F%3A%20boolean,-immediate%3F%3A

**Comment by RihanArfan** [comment]
See https://nuxt.com/docs/getting-started/data-fetching#lazy

In short, you can use `{ lazy: true }` as an option to useFetch, or alternatively use `useLazyFetch` instead of `useFetch`, which does the same thing.

Also, make sure you are not awaiting useFetch. 

```html
<script setup lang="ts">
// method 1
const { pending, data: posts } = useFetch('/api/posts', {
  lazy: true
})

// method 2
const { pending, data: posts } = useLazyFetch('/api/posts')
</script>

<template>
  <div>
    <p v-if="pending">Loading data...</p>
    <p v-else>{{ posts }}</p>
  </div>
</template>
```
